packages/react/src/components/Cards/CardGroup.props.ts

```diff
@@ -1,12 +1,31 @@
-import { CardProps } from "./Card.props";
+import {
+  DataCardProps,
+  DetailCardProps,
+  FactlistCardProps,
+  FeatureCardProps,
+  MultilinkCardProps,
+  PromoCardProps,
+  StatCardProps,
+  TextCardProps,
+} from "./Card.props";
 import { LinkProps } from "../LinkList/LinkList.props";
+import { CardTypes } from "../../types";
+
+type CardProps =
+  | DataCardProps
+  | DetailCardProps
+  | FactlistCardProps
+  | FeatureCardProps
+  | MultilinkCardProps
+  | PromoCardProps
+  | StatCardProps
+  | TextCardProps;
 
 export interface CardGroupProps {
   /**
    * An array of card objects
    */
-  cards?: Required<CardProps>[];
-
+  cards: CardProps[];
```

Comment:
This type declaration will also allow different card types to be used at once. The CardGroup is only supposed to be used with a single type of card at once right?

Reply:
Yes @bashlk, I added a union type here to make sure that the mapper card component will have one of the above types any suggestions ?

Reply:
I mean something like this

```
export interface CommonCardGroupProps {
  cardCount: Required<string>;
  cta?: LinkProps;
}

export interface StatCardGroup extends CommonCardGroupProps {
  type: "stat";
  cards: StatCardProps[];
}

export interface MultilinkCardGroup extends CommonCardGroupProps {
  type: "multilink";
  cards: MultilinkCardProps[];
}

...

export type CardGroupProps =
  | StatCardGroup
  | MultilinkCardGroup
  | TextCardGroup
  | PromoCardGroup
  | FeatureCardGroup
  | DetailCardGroup
  | FactlistCardGroup
  | DataCardGroup;
```

With this, someone won't be able to provide cards of a different type than the one specified as the type to the CardGroup.

Reply:
Also note that card count right now is not camel case. It probably should be. 

Also I don't think it is required at all. We can count the number of items passed into the cards prop to get the number of cards.

Reply:
The cardCount here refers to the amount of cards per row @bashlk. I think it makes sense to refactor the name of that variable right ?